### DISCUSSION TITLE
TypeScript repo classified as JavaScript

### DISCUSSION THREAD

**Discussion by Nixinova** [github-linguist/linguist#5249] [Classification] [answered]
I rewrote [my repo](https://github.com/Nixinova/pack-format) in TypeScript today, but my `.ts` files are showing up as JavaScript even after being reindexed.

![image](https://user-images.githubusercontent.com/42429413/109921327-9197c400-7d20-11eb-83dd-c425959f988d.png)

When rewriting the files I used `git mv index.js index.ts`; maybe that somehow messed with the classifier?

**Comment by lildude** [comment, marked as answer]
> maybe that somehow messed with the classifier?

Nope. This is the Search behaviour playing up and not Linguist (you should see that it is only the search results showing this). This is the same issue that affects https://github.com/github/linguist/issues/5052 and your previous discussion at https://github.com/github/linguist/discussions/5220

I opened an internal issue off your last discussion and I've since learnt the reason for this: the search functionality was written in Go and is using an old fork of https://github.com/src-d/enry taken before it was forked off into the publicly maintained https://github.com/go-enry/go-enry.  Our older fork has changes equivalent to Linguist v7.2.0 so is missing a whole lot of the newer languages etc.

Work is underway to get our changes up-streamed and then we'll switch over to using it instead at which point Search will start reflecting what the language bar does. There's no timeline on when this will happen though.

**Reply by Nixinova** [reply]
The language bar is also not showing TypeScript, though, so it is Linguist.

**Reply by lildude** [reply]
🤦 I missed that. Looking at the file, it's because of your shebang:

```
#!/usr/bin/env node
```

`node` is associated with Javascript:

https://github.com/github/linguist/blob/76c397c7ad07319e766fb1445b8a2307af2f586b/lib/linguist/languages.yml#L2655-L2666

... _not_ Typescript:

https://github.com/github/linguist/blob/76c397c7ad07319e766fb1445b8a2307af2f586b/lib/linguist/languages.yml#L5819-L5821

**Reply by Nixinova** [reply]
Huh, interesting. Looking at [this search query](https://github.com/search?q=extension%3Ats+language%3Ajs&type=code) I'm not the only one who gets their repo misclassified when you add the Node shebang. Which is valid syntax, by the way (https://github.com/Microsoft/TypeScript/issues/2749), and is the only way I can see to get my TS scripts to work in Node when compiled without adding any overhead. It's even [encouraged](https://itnext.io/how-to-create-your-own-typescript-cli-with-node-js-1faf7095ef89), and projects like [this](https://github.com/skovy/typed-scss-modules/blob/f006f40b4ecf7350ac8bf6da341ee20924f4f866/lib/cli.ts) are forced to use the shebang as otherwise the file cannot be run as a CLI. Should I make a PR adding node to the TS interpreters? This is very common and in the cases of making a CLI, the only easy option.

**Reply by Nixinova** [reply]
Since this is an issue I'll reopen this as one for better tracking.
#5251
